File: lib/java/openlinktoken/src/main/java/org/openlinktoken/tokens/tokenizer/SHA256TokenDigest.java
/* SPDX-License-Identifier: MIT */
package org.openlinktoken.tokens.tokenizer;

import java.security.MessageDigest;
import java.security.NoSuchAlgorithmException;

/**
 * Calculates SHA-256 token digests.
 */
public final class SHA256TokenDigest implements TokenDigest {

    /**
     * Calculates a SHA-256 digest.
     *
     * @param value the token-signature bytes
     * @return the SHA-256 digest
     * @throws NoSuchAlgorithmException if SHA-256 is unavailable
     */
    @Override
    public byte[] digest(byte[] value) throws NoSuchAlgorithmException {
        return MessageDigest.getInstance("SHA-256").digest(value);
    }
}

File: lib/python/openlinktoken/src/main/openlinktoken/tokens/tokenizer/sha256_token_digest.py
# SPDX-License-Identifier: MIT
"""SHA-256 token digest implementation."""

import hashlib

from openlinktoken.tokens.tokenizer.token_digest import TokenDigest


class Sha256TokenDigest(TokenDigest):
    """Calculate a SHA-256 token digest."""

    def digest(self, value: bytes) -> bytes:
        """Return the SHA-256 digest of the supplied bytes."""
        return hashlib.sha256(value).digest()
